src/components/reset-password-check-email/reset-password-check-email-controller.ts

```diff
@@ -51,12 +51,16 @@ export function resetPasswordCheckEmailGet(
       );
     }
 
-    if (result.success && req.session.user.enterEmailMfaType === undefined) {
-      req.session.user.enterEmailMfaType = result.data.mfaMethodType;
-      req.session.user.mfaMethods = upsertDefaultSmsMfaMethod(
-        req.session.user.mfaMethods,
-        { redactedPhoneNumber: result.data.phoneNumberLastThree }
-      );
+    if (result.success) {
+      req.session.user.mfaMethods = result.data.mfaMethods;
+
+      if (req.session.user.enterEmailMfaType === undefined) {
+        req.session.user.enterEmailMfaType = result.data.mfaMethodType;
+        req.session.user.mfaMethods = upsertDefaultSmsMfaMethod(
+          req.session.user.mfaMethods,
+          { redactedPhoneNumber: result.data.phoneNumberLastThree }
+        );
+      }
```

Comment:
I think we could drop the condition here and always update `enterEmailMfaType` here.

I also don't think we need to override the `req.session.user.mfaMethods` as this implementation with upsert was just a placeholder until we could get the API returning it as `result.data.mfaMetods` as above